.github/workflows/sdk-publish.yaml

```diff
@@ -126,15 +126,26 @@ jobs:
         uses: actions/setup-python@v4
         with:
           python-version: "3.8"
-      - name: Install dependencies
+      - name: Check for publish.sh
+        id: check-publish
         run: |
-          python -m pip install --upgrade pip
-          pip install setuptools wheel twine
-      - name: Build and publish
+          if [ -f scripts/publish.sh ]; then
+            echo "publish_with_makefile=true" >> $GITHUB_OUTPUT
```

Comment:
In general I think it's better to do this kind of check in the actual go code instead of in the GH action code. For example, we do checks for and set outputs like `publish_python` and `use_maven_legacy` (for Java) in the action code itself. Of course it's not really going to be less code or anything like that, I just think we should bias towards managing all of these things in a centralized place

I don't feel super strongly in this particular case, but thought I'd call it out.

Reply:
I bring up the `use_maven_legacy` thing because the result of it is quite similar--branching in the corresponding `publish` step

Reply:
I added this here because it was frankly easier to do and less wiring and I hope to deprecate it at some point if you want me to change it let me know otherwise I’ll leave it as is